pyccel/parser/semantic.py

```diff
@@ -3862,23 +3862,34 @@ def _visit_ClassDef(self, expr):
         methods = list(expr.methods)
         init_func = None
 
+        if not any(method.name == '__init__' for method in methods):
+            argument = FunctionDefArgument(Variable(cls.name, 'self', cls_base = cls))
+            scope = self.create_new_function_scope('__init__')
+            init_func = FunctionDef('__init__', [argument], (), [Pass()], cls_name=cls.name, scope=scope)
+            self.exit_function_scope()
```

Comment:
Is a new scope needed even though the `__init__` function is empty?

Reply:
Yes. All `ScopedAstNode`s should have a scope even if it is empty. It will be accessed when printing the `FunctionDef` to check if there are any local variables.

Also, the `__init__` function is empty here, but it will be modified here: https://github.com/pyccel/pyccel/blob/devel/pyccel/parser/semantic.py#L3867-L3882
to initialise the `is_freed` class variable so it is not permanently empty.